File: CS/CustomRowInGrid/PartContainer.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.ComponentModel;

namespace GridCustomRows
{
    class PartContainer : INotifyPropertyChanged
    {
        int _id;
        int _part1;
        int _part2;
        int _part3;


        public PartContainer()
        {
            _id = 0;
            _part1 = 0;
            _part2 = 0;
            _part3 = 0;
        }
        public PartContainer(int id, int part1, int part2, int part3)
        {
            _id = id;
            _part1 = part1;
            _part2 = part2;
            _part3 = part3;
        }
        public int ID
        {
            get { return _id; }
            set
            {
                if (
                    _id == value)
                    return;
                _id = value;
                RisePropertyChanged("ID");
            }
        }
        public int Part1
        {
            get { return _part1; }
            set
            {
                if (
                    _part1 == value)
                    return;
                _part1 = value;
                RisePropertyChanged("Part1");
            }
        }
        public int Part2
        {
            get { return _part2; }
            set
            {
                if (
                    _part2 == value)
                    return;
                _part2 = value;
                RisePropertyChanged("Part2");
            }
        }
        public int Part3
        {
            get { return _part3; }
            set
            {
                if (
                    _part3 == value)
                    return;
                _part3 = value;
                RisePropertyChanged("Part3");
            }
        }


        public event PropertyChangedEventHandler PropertyChanged;

        void RisePropertyChanged(string name)
        {
            if (PropertyChanged != null)
                PropertyChanged(this, new PropertyChangedEventArgs(name));

        }
    }

}

File: VB/CustomRowInGrid/PartContainer.vb
Imports System.ComponentModel

Namespace GridCustomRows

    Friend Class PartContainer
        Implements INotifyPropertyChanged

        Private _id As Integer

        Private _part1 As Integer

        Private _part2 As Integer

        Private _part3 As Integer

        Public Sub New()
            _id = 0
            _part1 = 0
            _part2 = 0
            _part3 = 0
        End Sub

        Public Sub New(ByVal id As Integer, ByVal part1 As Integer, ByVal part2 As Integer, ByVal part3 As Integer)
            _id = id
            _part1 = part1
            _part2 = part2
            _part3 = part3
        End Sub

        Public Property ID As Integer
            Get
                Return _id
            End Get

            Set(ByVal value As Integer)
                If _id = value Then Return
                _id = value
                RisePropertyChanged("ID")
            End Set
        End Property

        Public Property Part1 As Integer
            Get
                Return _part1
            End Get

            Set(ByVal value As Integer)
                If _part1 = value Then Return
                _part1 = value
                RisePropertyChanged("Part1")
            End Set
        End Property

        Public Property Part2 As Integer
            Get
                Return _part2
            End Get

            Set(ByVal value As Integer)
                If _part2 = value Then Return
                _part2 = value
                RisePropertyChanged("Part2")
            End Set
        End Property

        Public Property Part3 As Integer
            Get
                Return _part3
            End Get

            Set(ByVal value As Integer)
                If _part3 = value Then Return
                _part3 = value
                RisePropertyChanged("Part3")
            End Set
        End Property

        Public Event PropertyChanged As PropertyChangedEventHandler Implements INotifyPropertyChanged.PropertyChanged

        Private Sub RisePropertyChanged(ByVal name As String)
            RaiseEvent PropertyChanged(Me, New PropertyChangedEventArgs(name))
        End Sub
    End Class
End Namespace
